File: equipment-svc/Equipment.Domain/Entities/Department.cs
namespace Equipment.Domain.Entities;

public class Department : BaseEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int? ManagerId { get; set; }
    public bool IsActive { get; set; } = true;
}

File: equipment-app/src/types/Department.ts
export interface DepartmentEntity {
    id: number,
    code: string,
    name: string,
    description: string,
    managerId: number,
    managerName: string,
    isActive: boolean,
}

export interface DepartmentModel {
    id: number,
    code: string,
    name: string,
}

export interface DepartmentPaging {
    id: number,
    code: string,
    name: string,
    quantityEquipment: number,
    quantityUser: number,
    managerName: string,
    isActive: boolean,
    createdDate: Date,
    updatedDate: Date,
}

export interface DepartmentResponseModel {
    isSuccess: boolean;
    codeError: string;
}